registry/storage/driver/s3-aws/s3.go

```diff
@@ -874,6 +875,139 @@ func (d *driver) URLFor(ctx context.Context, path string, options map[string]int
 	return req.Presign(expiresIn)
 }
 
+// Walk traverses a filesystem defined within driver, starting
+// from the given path, calling f on each file
+func (d *driver) Walk(ctx context.Context, from string, f storagedriver.WalkFn) error {
+	var objectCount int64
+
+	path := from
+	if path != "/" && path[len(path)-1] != '/' {
+		path = path + "/"
+	}
+
+	prefix := ""
+	if d.s3Path("") == "" {
+		prefix = "/"
+	}
+
+	if err := d.doWalk(ctx, &objectCount, d.s3Path(path), prefix, f); err != nil {
+		return err
+	}
+
+	if objectCount == 0 {
```

Comment:
Could you leave a comment why this errors? Does this mean it'll error on an empty directory?

Reply:
The idea of an "empty" directory doesn't exist on EC2, see:
```
(titusagent) ~ $ aws s3 ls s3://bucket/titan/devvpc7/
                           PRE logs/
                           PRE registry/
                           PRE t/

(titusagent) ~ $ aws s3 cp foo s3://bucket/titan/devvpc7/sargun/
upload: ./foo to s3://bucket/titan/devvpc7/sargun/foo

(titusagent) ~ $ aws s3 ls s3://bucket/titan/devvpc7/
                           PRE logs/
                           PRE registry/
                           PRE sargun/
                           PRE t/

(titusagent) ~ $ aws s3 rm s3://bucket/titan/devvpc7/sargun/foo
delete: s3://bucket/titan/devvpc7/sargun/foo

(titusagent) ~ $ aws s3 ls s3://bucket/titan/devvpc7/
                           PRE logs/
                           PRE registry/
                           PRE t/
```